packages/shared/src/components/auth/RegistrationForm.tsx

```diff
@@ -0,0 +1,164 @@
+import classNames from 'classnames';
+import React, {
+  MutableRefObject,
+  ReactElement,
+  useEffect,
+  useState,
+} from 'react';
+import { formInputs } from '../../lib/form';
+import { Button } from '../buttons/Button';
+import ImageInput from '../fields/ImageInput';
+import { TextField } from '../fields/TextField';
+import MailIcon from '../icons/Mail';
+import UserIcon from '../icons/User';
+import VIcon from '../icons/V';
+import { CloseModalFunc } from '../modals/common';
+import AuthModalHeader from './AuthModalHeader';
+import { AuthForm } from './common';
+import EmailVerificationSent from './EmailVerificationSent';
+
+export interface SocialProviderAccount {
+  provider: string;
+  name: string;
+  image?: string;
+}
+
+interface RegistrationFormProps {
+  email: string;
+  socialAccount?: SocialProviderAccount;
+  formRef?: MutableRefObject<HTMLFormElement>;
+  onClose?: CloseModalFunc;
+  onBack?: CloseModalFunc;
+  isV2?: boolean;
+}
+
+export interface RegistrationFormValues {
+  image?: string;
+  email?: string;
+  fullname?: string;
+  password?: string;
+  username?: string;
+}
+
+export const RegistrationForm = ({
+  email,
+  socialAccount,
+  formRef,
+  onClose,
+  onBack,
+  isV2,
+}: RegistrationFormProps): ReactElement => {
+  const [isNameValid, setIsNameValid] = useState<boolean>();
```

Comment:
A general question - haven't we ever thought about having a form-handling solution? Probably this is something we can discuss on our web-team meeting @rebelchris

Reply:
Good point. Was thinking about that as well. My first option is `react-forms`.

Reply:
It's something we will probably research if we decided to go with some solution - have never heard `react-forms`, but `react-hook-form` is a great choice from my experience (zero dependencies, small and fast)

Reply:
I might have forgotten the name, but I think, yes, that's the same I'm referring to.

Reply:
We have a built-in one to help us with forms, I haven't checked which parts can be salvaged, I suggest we put it in another ticket as researching for options and which one to use would take time. Also migrating the existing implementations would require work. Though if we have time, then I think we can take it on in this same shipment.